docs/source/howto/codes.rst

```diff
@@ -4,41 +4,265 @@
 
 Design guidelines
-------------------
+-----------------
 
+ * | **Check existing resources.**
+   | Before starting to write a plugin, check on the `aiida plugin registry <https://aiidateam.github.io/aiida-registry/>`_ whether a plugin for your code is already available.
  * | **Start simple.**
    | Make use of existing classes like :py:class:`~aiida.orm.nodes.data.dict.Dict`, :py:class:`~aiida.orm.nodes.data.singlefile.SinglefileData`, ...
    | Write only what is necessary to pass information from and to AiiDA.
  * | **Don't break data provenance.**
    | Store *at least* what is needed for full reproducibility.
- * | **Parse what you want to query for.**
-   | Make a list of which information to:
-
-     #. parse into the database for querying (:py:class:`~aiida.orm.nodes.data.dict.Dict`, ...)
-     #. store in the file repository for safe-keeping (:py:class:`~aiida.orm.nodes.data.singlefile.SinglefileData`, ...)
-     #. leave on the computer where the calculation ran (:py:class:`~aiida.orm.nodes.data.remote.RemoteData`, ...)
-
  * | **Expose the full functionality.**
    | Standardization is good but don't artificially limit the power of a code you are wrapping - or your users will get frustrated.
    | If the code can do it, there should be *some* way to do it with your plugin.
-
  * | **Don't rely on AiiDA internals.**
    | AiiDA's :ref:`public python API<python_api_public_list>` includes anything that you can import via ``from aiida.module import thing``.
    | Functionality at deeper nesting levels is not considered part of the public API and may change between minor AiiDA releases, breaking your plugin.
 
+.. _how-to:codes:parsing:
+
+Parsing the outputs
+===================
+
+The parsing step occurs after the calculation has finished running and all the relevant outputs have been retrieved.
+It is an optional step that allows you to extract relevant information from the output files and store it into AiiDA nodes in formats that are easier and quicker to query and analyze.
+The way to implement this is by extending the base |Parser| class as you can see below:
+
+.. code-block:: python
+   :linenos:
+
+    from aiida.parsers.parser import Parser
+    from aiida import orm
+
+    class ArithmeticAddParser(Parser):
+
+        def parse(self, **kwargs):
+
+            output_folder = self.retrieved
+            output_filename = self.node.get_attribute('output_filename')
+
+            with output_folder.open(output_filename, 'r') as handle:
+                result = int(handle.read())
+
+            self.out('sum', orm.Int(result))
+
+The first command in the example (line 8) shows the way to get the ``retrieve`` folder that was generated by the associated |CalcJob| and that, by this point, should contain the files included in the ``retrieve_list`` (as specified in the |prepare_for_submission| method).
```

Comment:
The `| prepare_for_submission|` reference does not resolve properly.
https://88-77234579-gh.circle-artifacts.com/0/html/howto/codes.html#parsing-the-outputs